python/python/tests/test_pubsub.py

```diff
@@ -167,29 +233,31 @@ async def test_pubsub_exact_happy_path(
         Async, Sync, and Callback. It verifies that a message published to a
         specific channel is correctly received by a subscriber.
         """
-        channel = get_random_string(10)
-        message = get_random_string(5)
-
-        callback, context = None, None
-        callback_messages: List[CoreCommands.PubSubMsg] = []
-        if method == MethodTesting.Callback:
-            callback = new_message
-            context = callback_messages
-
-        pub_sub = create_pubsub_subscription(
-            cluster_mode,
-            {GlideClusterClientConfiguration.PubSubChannelModes.Exact: {channel}},
-            {GlideClientConfiguration.PubSubChannelModes.Exact: {channel}},
-            callback=callback,
-            context=context,
-        )
+        listening_client, publishing_client = None, None
+        try:
+            channel = get_random_string(10)
+            message = get_random_string(5)
+
+            callback, context = None, None
+            callback_messages: List[CoreCommands.PubSubMsg] = []
+            if method == MethodTesting.Callback:
+                callback = new_message
+                context = callback_messages
+
+            pub_sub = create_pubsub_subscription(
+                cluster_mode,
+                {GlideClusterClientConfiguration.PubSubChannelModes.Exact: {channel}},
+                {GlideClientConfiguration.PubSubChannelModes.Exact: {channel}},
+                callback=callback,
+                context=context,
+            )
 
-        publishing_client, listening_client = await create_two_clients(
-            request, cluster_mode, pub_sub
-        )
+            listening_client, publishing_client = await create_two_clients_with_pubsub(
+                request, cluster_mode, pub_sub
+            )
 
-        try:
             result = await publishing_client.publish(message, channel)
+            # TODO: enable when client closing works
```

Comment:
enable what?
and client close is now closing immediately

Reply:
removed